homeassistant/components/nut/__init__.py

```diff
@@ -180,27 +193,34 @@ def __init__(self, host, port, alias, username, password):
         # Establish client with persistent=False to open/close connection on
         # each update call.  This is more reliable with async.
         self._client = PyNUTClient(self._host, port, username, password, 5, False)
+        self._ups_list = {}
         self._status = None
 
     @property
     def status(self):
         """Get latest update if throttle allows. Return status."""
-        self.update()
         return self._status
 
     @property
     def name(self):
         """Return the name of the ups."""
         return self._alias
 
-    def list_ups(self):
+    @property
+    def ups_list(self):
+        """Return the list of upses."""
+        return self._ups_list
+
+    def _update_ups_list(self) -> dict:
         """List UPSes connected to the NUT server."""
-        return self._client.list_ups()
+        self._ups_list = self._client.list_ups()
+        return self._ups_list
 
     def _get_alias(self):
         """Get the ups alias from NUT."""
         try:
-            return next(iter(self.list_ups()))
+            ups_list = self._update_ups_list()
+            return list(ups_list.keys())[0]
```

Comment:
Copying a dict to a list copies the keys by default.
```suggestion
            return list(ups_list)[0]
```

Reply:
done